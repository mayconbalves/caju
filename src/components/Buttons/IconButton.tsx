import { IconButton as WrapperButton } from './styles'
import { IconButtonProps } from './types'

const IconButton = ({ children, onClick }: IconButtonProps) => {
  return (
    <WrapperButton role="button" onClick={onClick}>
      {children}
    </WrapperButton>
  )
}

export default IconButton
