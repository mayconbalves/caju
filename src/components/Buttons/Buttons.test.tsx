import { fireEvent, render, screen } from '@testing-library/react'
import Buton from './Button'
import IconButton from './IconButton'

describe('Component Buttons', () => {
  describe('Buton Component', () => {
    it('renders with children text', () => {
      render(
        <Buton bgcolor="#64a98c" type="button" onClick={jest.fn()}>
          Click Me
        </Buton>
      )

      const button = screen.getByRole('button', { name: /click me/i })
      expect(button).toBeInTheDocument()
    })

    it('calls onClick when clicked', () => {
      const mockOnClick = jest.fn()
      render(
        <Buton bgcolor="#64a98c" type="button" onClick={mockOnClick}>
          Click Me
        </Buton>
      )

      const button = screen.getByRole('button', { name: /click me/i })
      fireEvent.click(button)

      expect(mockOnClick).toHaveBeenCalledTimes(1)
    })

    it('renders with correct background and text color', () => {
      render(
        <Buton bgcolor="blue" color="white" onClick={jest.fn()}>
          Small Button
        </Buton>
      )

      const button = screen.getByRole('button', { name: /small button/i })
      expect(button).toBeInTheDocument()
      expect(button).toHaveStyle(`background-color: blue`)
      expect(button).toHaveStyle(`color: white`)
    })

    it('calls onClick when clicked', () => {
      const mockOnClick = jest.fn()
      render(
        <Buton bgcolor="blue" color="white" onClick={mockOnClick}>
          Small Button
        </Buton>
      )

      const button = screen.getByRole('button', { name: /small button/i })
      fireEvent.click(button)

      expect(mockOnClick).toHaveBeenCalledTimes(1)
    })
  })

  describe('IconButton Component', () => {
    it('renders with children', () => {
      render(
        <IconButton onClick={jest.fn()}>
          <span>Icon</span>
        </IconButton>
      )

      const button = screen.getByRole('button')
      expect(button).toBeInTheDocument()
      expect(screen.getByText(/icon/i)).toBeInTheDocument()
    })

    it('calls onClick when clicked', () => {
      const mockOnClick = jest.fn()
      const { container } = render(
        <IconButton onClick={mockOnClick}>
          <span>Icon</span>
        </IconButton>
      )

      const button = screen.getByRole('button')
      fireEvent.click(button)

      expect(mockOnClick).toHaveBeenCalledTimes(1)
      expect(container).toMatchSnapshot()
    })
  })
})
