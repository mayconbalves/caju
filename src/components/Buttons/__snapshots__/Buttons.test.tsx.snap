// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Buttons IconButton Component calls onClick when clicked 1`] = `
<div>
  <button
    class="sc-gEvDqW jzmXzC"
    role="button"
  >
    <span>
      Icon
    </span>
  </button>
</div>
`;
